### PR TITLE
Window: Add debug console

### LINKED ISSUE #84: Better debugging through an in-window console
While debuggers are a fantastic tool, it is often better to use simple `print` debugging; however, because we use ncurses for the TUI, print statements are not useable in zygrader for this purpose. A good solution would be to add a console to the zygrader window that can be enabled at startup with a flag. This console could display below the main zygrader window.

* Enabled with with a `-d | --debug` flag.
* To print to the debug console, use `window.debug()` which accepts a string to print

It could look like the following
```
+------------------+
|main window       |
|                  |
|                  |
|                  |
|                  |
+------------------+
|log info here     |
|                  |
+------------------+
```

Keeping the window at the bottom would make implementation simpler. Any time the window height is queried we can just subtract `n` rows from the height. Then the window manager can just draw text below the main zygrader window.

The console would just log statements printed by `window.debug()`, scrolling with each line printed.

---

Additional ideas
* Perhaps an additional flag to log all messages printed to the debug window to a file

### PR DESCRIPTION
**Pull request by natecraddock**
Adds a debugging console to the window manager. This can be enabled with the `-d` or `--debug` flags to zygrader.

At the moment the height is set to 4 rows and will scroll to show the last 4 rows of output. At this point it is useful enough to include in zygrader, but more could be added in the future.

### CODE REVIEW

File: zygrader/ui/window.py [hunk 1]
```diff
@@ -265,6 +284,23 @@ def draw(self):
...
         curses.doupdate()
 
+    def draw_debug_console(self):
+        self.__debug_win.erase()
+
+        # First row is a separator line
+        self.__debug_win.hline(0, 0, "_", self.cols)
+
+        row_num = 1
+        display_rows = self.__debug_lines[-4:]
```
Comment (teiknsmith) [inline zygrader/ui/window.py]: should probably make use of the height constant

**Review by teiknsmith — CHANGES_REQUESTED** [review]

**Comment by natecraddock** [comment]
Was there anything else here you wanted to look at, or am I good to commit now?

**Comment by spencerteiknsmith** [comment]
that was all - although the behavior is going to be changed with that cleanup. the value was 4 before, and the constant is defined as 5. I'm not sure if that was intentional and fixes behavior, or if it should be `-(DEBUG_CONSOLE_HEIGHT - 1)`. I'll trust you to check that and make sure everything is as it should be

**Review by teiknsmith — COMMENTED** [review]
that was all - although the behavior is going to be changed with that cleanup. the value was 4 before, and the constant is defined as 5. I'm not sure if that was intentional and fixes behavior, or if it should be -(DEBUG_CONSOLE_HEIGHT - 1). I'll trust you to check that and make sure everything is as it should be

**Comment by natecraddock** [comment]
Derp, you're right. The original constant was 4, so that's just a stupid mistake. Thanks!

**Pull request closed without merging**